File: backend/java/localpos-product-service/src/main/java/com/pds/localpos/productservice/config/I18nConfigurationProperties.java
package com.pds.localpos.productservice.config;

import lombok.Getter;
import lombok.Setter;
import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.stereotype.Component;

import java.util.Locale;

@Component
@ConfigurationProperties(prefix = "app")
@Getter
@Setter
public class I18nConfigurationProperties {

    private String locale;

    public Locale getLocale() {
        return (locale != null && !locale.isBlank()) ? Locale.forLanguageTag(locale) : null;
    }
}

File: backend/kotlin/localpos-user-service-kt/src/main/kotlin/com/pds/localpos/userservice/config/I18nConfigurationProperties.kt
package com.pds.localpos.userservice.config

import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.stereotype.Component
import java.util.*

@Component
@ConfigurationProperties(prefix = "app")
class I18nConfigurationProperties {

    var locale: String? = null

    val resolvedLocale: Locale
        get() = if (!locale.isNullOrBlank()) Locale.forLanguageTag(locale) else Locale.getDefault()
}
